### ISSUE TITLE
closing child window does not always focus parent

### ISSUE THREAD

**Issue by aeosynth** [JLErvin/berry#72]
The most recently opened window is focused when a window closes, instead of the parent / spawner of the closed window.

example: open two terminals, use each to open an image. the newest terminal is always focused after the image closes, instead of the terminal that spawned the image.

**Comment by JLErvin** [comment]
`berry` is a non-reparenting window manager, similar to `katriawm`. It's unclear to me whether or not this is feasible to implement, although I can look into it.

**Comment by JLErvin** [comment]
So, I have developed an imperfect but reasonable solution on the development branch. Until now, `berry` has default to focusing the _next_ window in the focus list whenever a client is closed. However, if we instead focus the _previous_ window, we achieve this behavior about 99% of the time. I see this situation being most common when you do some action in an application that spawns another window (i.e. a file browser to send an email attachment, or something similar). Since the parent window is _almost_ always focused last, we revert to that window, emulating parent focusing without needing re-parenting. 

Such a solution is imperfect in a few edge cases. If you were to 1) open a popup 2) open another new window and switch to it 3) close the popup, focus would revert to the newly opened window, not the parent of the popup. Likewise, if a window spawns a child without being focused this would not work either. However, both of these situations seems rare and not something I see as a high priority. 

The feature is available on the development branch [here](https://github.com/JLErvin/berry/tree/huckleberry). Feel free to try it out. I will close this issue once the feature has been merged to master. 

:)
